validateFileNames.js

```diff
@@ -0,0 +1,94 @@
+const shelljs = require('shelljs');
+
+// Regex that defines our file and directory naming convention
+const camelCasedFileNameRegex = /^[.]?([a-z])+([0-9]|[a-zA-Z]|[.])*$/;
+
+// Files and directories that are exempt from the naming convention
+const ignoredFilesAndDirectories = [
+  'README.md',
+  'Dockerfile',
+  'LICENSE.md',
+  'customTypings',
+  'typings',
+];
+
+// Get the list of files that we're interested in validating
+const files = getFiles();
+
+// Create a place to store files in violation of naming convention
+const filesInViolation = [];
+
+// Now let's go through the files to see if they violate the naming convention
+files.forEach(file => {
+  // Split the path of the file to get the path components
+  const pathComponents = file.split('/');
```

Comment:
Would that work on Windows? Paths in Windows are split with a backslash (`\`). There must be a Node.js API or a 3rd-party module to get the path components.

Edit: `path.sep` returns the platform-specific separator. [Node.js documentation](https://nodejs.org/api/path.html#path_path_sep) has an example on how to split using `path.sep`.